### ISSUE TITLE
Meteor stuck at "scanning local packages"

### ISSUE THREAD

**Issue by znewsham** [meteor/meteor#7670]
version 1.4.1, previously version 1.4.0

Since the upgrade to 1.4 I've been having intermittent issues running meteor. The latest in this is that after making any change to the code I get "Scanning local packages" which remains forever

I've tried doing a new install of meteor, and resetting the project - neither of these work. 

I can run the project, but not make any changes while running.

**Comment by benjamn** [comment]
Do you have a local `packages/` directory in your app?

**Comment by wilau2** [comment]
@benjamn  hi i currently have problem with updating to meteor 1.4.1 with my local packages/ 
test:test
it creates a test_test folder then i have error bad path 

Error: bad path: ../sambegin_push/.npm/plugin/configuration/node_modules
    at rejectBadPath (/tools/isobuild/bundler.js:185:11)
    at NodeModulesDirectory.getPreferredBundlePath (/tools/isobuild/bundler.js:273:5)
    at /tools/isobuild/bundler.js:1740:39

**Comment by maximchouinard** [comment]
+1

**Comment by badmark** [comment]
Also experiencing this specifically after updating to 1.4.1. I do have local packages due to the fact I need to build them locally. Project will run fine until I make a code change, then it just hangs at "scanning local packages" requiring me to stop and restart the Meteor process.

Why are local packages causing this?

**Comment by benjamn** [comment]
@wilau2 That doesn't have anything to do with this issue. Please [open a new issue](https://github.com/meteor/meteor/issues/new) and make sure to follow the instructions in the issue template. Thanks.

**Comment by benjamn** [comment]
@badmark I don't have an explanation yet, sorry.

**Comment by znewsham** [comment]
@benjamn, I do have local packages - and they are required. 

This briefly (last night, but back again now) went away for me, but I'm not sure what fixed it. The steps I followed (not easily repeatable)

1) upgrade from 1.4.0 to 1.4.1
2) reinstall 1.4.1 using curl
3) meteor reset
4) create new project, copy code and packages in manually, adding where necessary
5) tried downgrading to 1.3.3

None of these directly did it, but last night after about 30 minutes I noticed while editing that I hadn't needed to restart the project between changes. I suspect there was an error somewhere in the code that wasn't being reported causing it to lock up

I just kicked off the project again and made a change, and its back :(

In some cases, changes dont trigger any update at all (html, client side js or server side js)

**Comment by znewsham** [comment]
It looks like if I change html first, no other changes are recognised
If I change js first, I get "scanning local packages" forever

**Comment by badmark** [comment]
I can reproduce @znewsham  JS changes trigger hang.

**Comment by znewsham** [comment]
@badmark do you need any other information from me?

**Comment by mrmowgli** [comment]
Same here: Git clone the simple-todos app, change into directory, meteor npm install works.

running meteor results in a hang downloading the update tool.
`Downloading meteor-tool@1.4.1...          [==========                ] 37% 52.1s`

goes to the end and the progress bar disappears but it still says it is downloading and never continues.

It just stays stuck at: 
`Downloading meteor-tool@1.4.1...          /`

**Comment by laosb** [comment]
@mrmowgli Your case should totally be reported to #6960 with some extra info described there.

**Comment by znewsham** [comment]
both issues still present with 1.4.1.1

**Comment by znewsham** [comment]
This appears to be a linux only issue (a mac user doesn't see the same issue)

**Comment by abernix** [comment]
@znewsham Any flavour of Linux in particular?

If you (or anyone) could come up with some easy reproduction steps and a Meteor project GitHub repo (with dependencies, `packages/`, etc.) it would be most helpful, otherwise this becomes hard to debug/reproduce/fix for anyone who is not running into it.

**Comment by znewsham** [comment]
I'm running on Ubuntu 14.04 LTS - I actually have two machines with this setup, one "macbook" and one AMD desktop, they both exhibit similar issues (but not identical). I'll try to setup a github repo as the project in question does have some custom packages in it.

**Comment by dagatsoin** [comment]
I am running similar issue randomly. The workaround I do is the same as https://github.com/meteor/todos/issues/107#issuecomment-250779974
I reset meteor to delete the builds and run it again.

I have this issue since older version (1.2 as long I remember) and I have it actually with 1.4.1 on a Linux container (don't know which flavor) : running `meteor build /app` will hangs if I did not delete the previous destination folder.

**Comment by RobinKuiper** [comment]
I have the same thing on a Windows PC, but not all the time, only after some time running Meteor I get this.

**Comment by oliverlloyd** [comment]
It's also an issue with OSX (10.10.5). And yes, using custom local packages.
`Meteor 1.4.1.3`

**Comment by lexmustafin** [comment]
+1. had to 'meteor run' 
Meteor 1.4.1.3

**Comment by abernix** [comment]
@lexmustafin I'm not sure if you're saying that using `meteor run` fixed the problem for you but running `meteor` and `meteor run` are equivalent so that probably would have been the same as just pressing Ctrl-C and trying again.

**Comment by lexmustafin** [comment]
@abernix , right. that's what i meant, sorry. I had to press CMD+C and try again.

**Comment by ramezrafla** [comment]
Is this still an issue?

**Comment by dagatsoin** [comment]
Not for me since METEOR@1.4.2 but my project hits a nearly stable stage and. So this could not be revelant. Has something been done about this?

**Comment by ramezrafla** [comment]
Don't think so - trying to clear old issues. Thanks @dagatsoin  for responding.

**Comment by mrmowgli** [comment]
My project seems stable as well, I think you can go ahead and clise this.

**Comment by ramezrafla** [comment]
@hwillson, @abernix  can you do us the honors and close?

**Comment by laosb** [comment]
Glad to see your stable working project - closing!
